cores/esp8266/libc_replacements.cpp

```diff
@@ -127,6 +127,7 @@ void _exit(int status) {
     abort();
 }
 
+int atexit(void (*func)()) __attribute__((weak));
```

Comment:
I think this was mentioned before, weak definition will apply to every symbol otherwise. Placing it here we specify that libc_replacements.o version is weak

Reply:
I don't understand this comment. Please explain in detail.

Reply:
`__attribute__((weak));` when in header is applied to every file including Arduino.h and implementing the function, so every atexit becomes weak and we depend in which order they are linked.